Source/x11/XGServerEvent.m

```diff
@@ -1894,6 +1898,25 @@ - (void) processEvent: (XEvent *) event
                         ((XShmCompletionEvent *)&xEvent)->drawable];
             break;
           }
+#endif
+#ifdef HAVE_XRANDR
+        int randr_event_type = randrEventBase + RRScreenChangeNotify;
+        if (xEvent.type == randr_event_type
+            && (xEvent.xconfigure.window == RootWindow(dpy, defScreen)))
+          {
+            // Check if other RandR events are waiting in the queue.
+            XSync(dpy,0);
+            while (XCheckTypedEvent(dpy, randr_event_type, &xEvent)) {;}
+                    
+            XRRUpdateConfiguration(event);
+            // Regenerate NSScreens
+            [NSScreen resetScreens];
+            // Notify application about screen parameters change
```

Comment:
Should we move this notification into the NSScreen method?

Reply:
`resetScreens` is a GNUstep addition. I'm not sure about exact purpose of this method. Anyone can call it without actual screen change. Do you think it's safe?

Reply:
Why would code call that method without screen changes? What we should try to figure out is where that notification code belongs logically. If it is more likely that gui objects are listening for that notification I would prefer to have it there. That way you understand what is going on by just looking at the gui code. You won't have to dive into back to see the logic.

Reply:
> Why would code call that method without screen changes? 

For example when NSScreen or some application want to refresh screens information.

> What we should try to figure out is where that notification code belongs logically. 

I think this notification is at the right place - where screen changes (RandR) event processed.

> If it is more likely that gui objects are listening for that notification I would prefer to have it there. That way you understand what is going on by just looking at the gui code. You won't have to dive into back to see the logic.

I understand. But you have to dive into back code anyway because `DESTROY(screenArray)` line in `resetScreens` will not clear the situation up. You probably need to find out who have called `resetScreens`. From arhitectural point of view `resetScreens` can be replaced with some other method (screenArray readwrite property) because `resetScreens` has one purpose - force NSScreen to update its information about displays on next call to `screens`.